src/Datadog.Trace.ClrProfiler.Managed/Integrations/GraphQLIntegration.cs

```diff
@@ -0,0 +1,351 @@
+            object document,
+            object rules,
+            object userContext,
+            object inputs,
+            int opCode,
+            int mdToken)
+        {
+            if (documentValidator == null) { throw new ArgumentNullException(nameof(documentValidator)); }
+
+            const string methodName = nameof(Validate);
+
+            // At runtime, get a Type object for GraphQL.ExecutionResult
+            var documentValidatorInstanceType = documentValidator.GetType();
+            Type graphQLExecutionResultType;
+            Type documentValidatorInterfaceType;
+
+            try
+            {
+                var graphQLAssembly = AppDomain.CurrentDomain.GetAssemblies()
+                                        .Where(a => a.GetName().Name.Equals(GraphQLAssemblyName))
+                                        .Single();
+                graphQLExecutionResultType = graphQLAssembly.GetType(GraphQLExecutionResultName, throwOnError: true);
+                documentValidatorInterfaceType = graphQLAssembly.GetType(GraphQLDocumentValidatorInterfaceName, throwOnError: true);
+            }
+            catch (Exception ex)
+            {
+                // This shouldn't happen because the GraphQL assembly should have been loaded to construct various other types
+                // profiled app will not continue working as expected without this method
+                Log.ErrorException($"Error calling {documentValidatorInstanceType.Name}.{methodName}(IConnection connection, CancellationToken cancellationToken)", ex);
+                throw;
+            }
+
+            Func<object, object, object, object, object, object, object, object> instrumentedMethod;
+
+            try
+            {
+                instrumentedMethod =
+                    MethodBuilder<Func<object, object, object, object, object, object, object, object>>
+                        .Start(Assembly.GetCallingAssembly(), mdToken, opCode, methodName)
+                        .WithConcreteType(documentValidatorInstanceType)
+                        .WithParameters(originalQuery, schema, document, rules, userContext, inputs)
+                        .Build();
+            }
+            catch (Exception ex)
+            {
+                // profiled app will not continue working as expected without this method
+                Log.ErrorException($"Error resolving {documentValidatorInstanceType.Name}.{methodName}(IConnection connection, CancellationToken cancellationToken)", ex);
+                throw;
+            }
+
+            using (var scope = CreateScopeFromValidate(document))
+            {
+                try
+                {
+                    var validationResult = instrumentedMethod(documentValidator, originalQuery, schema, document, rules, userContext, inputs);
+
+                    // Mark the span as an error if the validation failed
+                    if (!validationResult.GetProperty<bool>("IsValid").GetValueOrDefault())
```

Comment:
if IsValid doesn't exist on `validationResult` the default will return as true. It might be more accurate to handle the fact that we couldn't find the property, rather than always reporting an error from GraphQL. It could be that it is renamed or moved in future versions and would be nice if it was called out explicitly.

Reply:
That's a good observation. I actually have a change that I haven't yet pushed where I just check for `Errors.Count > 0`. Since Count is an int, the default is 0 and if for some reason we cannot find those properties, we will not go into the error-logging logic.

Reply:
Fixed in 6451a89675d7b6cfad7c17660be44a5a4eb338a8 . Let me know if that works for you.